relayer-rest/src/server.rs

```diff
@@ -0,0 +1,52 @@
+use std::thread;
+
+use crossbeam_channel as channel;
+use tracing::{info, trace};
+
+use ibc_relayer::rest::request::Request;
+
+use crate::{
+    handle::{all_chain_ids, assemble_version_info, chain_config, supervisor_state},
+    Config,
+};
+
+pub fn spawn(config: Config) -> (thread::JoinHandle<()>, channel::Receiver<Request>) {
+    let (request_sender, request_receiver) = channel::unbounded::<Request>();
+    info!("[rest] starting REST API server at {}", config.address);
+    (
+        thread::spawn(move || run(config, request_sender)),
+        request_receiver,
+    )
+}
+
+#[allow(clippy::manual_strip)]
+fn run(config: Config, sender: channel::Sender<Request>) {
+    rouille::start_server(config.address, move |request| {
+        router!(request,
+            (GET) (/) => {
+                trace!("[rest/server] GET /");
+                let result = assemble_version_info(&sender);
+                rouille::Response::json(&result)
+            },
+
+            (GET) (/chain) => {
+                // TODO(Soares): Add a `into_detail` to consume the error and obtain
+                //   the underlying detail, so that we avoid doing `e.0`
+                trace!("[rest/server] GET /chain");
+                rouille::Response::json(&all_chain_ids(&sender).map_err(|e| e.0))
```

Comment:
@soareschen is this feasible with flex-error? We'd like to be able to do something as follows:

```suggestion
                trace!("[rest/server] GET /chain");
                rouille::Response::json(&all_chain_ids(&sender).map_err(|e| e.into_detail()))
```

The `into_detail` method should convert the error into the underlying `RestApiError` type (which derives `Serialize`) so we can output it in JSON format.

Reply:
Yes, we can include that in the macro generation: similar to the `detail()` method, which returns a reference: https://github.com/informalsystems/flex-error/blob/master/flex-error/src/macros.rs#L515-L517

Reply:
Sounds good, we'll keep this in mind for later. Thanks Soares!